### ISSUE TITLE
Unintelligible Pronunciation after Training from Scratch

### ISSUE THREAD

**Issue by w11wo** [SWivid/F5-TTS#509]
### Checks

- [X] This template is only for usage issues encountered.
- [X] I have thoroughly reviewed the project documentation but couldn't find information to solve my problem.
- [X] I have searched for existing issues, including closed ones, and couldn't find a solution.
- [X] I confirm that I am using English to submit this report in order to facilitate communication.

### Environment Details

`torch==2.3.0+cu121`

### Steps to Reproduce

Hi there, I'll try to detail as much as possible with my training setup and please let me know if I need to clarify some other things.

I'm training on a combination of English + Indonesian + Swahili audio, 44.1kHz. About 2h worth of data. I'm training from scratch with the following config:

```
target_sample_rate = 44100
n_mel_channels = 128
hop_length = 256
mel_spec_type = "bigvgan"
```

and am planning to use `nvidia/bigvgan_v2_44khz_128band_256x` since I cannot find 44.1kHz version of Vocos. I ran:

```sh
accelerate launch --num_processes=1 src/f5_tts/train/finetune_cli.py \
    --dataset_name en_id_sw \
    --learning_rate 2e-4 \
    --batch_size_per_gpu 3200 \
    --epochs 2300 \
    --save_per_updates 100000 \
    --last_per_steps 100000 \
    --tokenizer pinyin \
    --tokenizer_path data/en_id_sw/vocab.txt \
    --log_samples true
```

My `vocab.txt` looks like:

```
 
!
'
,
.
:
?
a
b
c
d
e
f
g
h
i
j
k
l
m
n
o
p
q
r
s
t
u
v
w
x
y
z
```

and I've trained my model for 1M steps.

Here's the sample inference output: https://voca.ro/15EqoXsQYxVG

The voice quality sounds great, but it's just the content that is completely unintelligible. The audio should've contained `the quick brown fox jumps.`.

I've double-checked with tokenization and confirmed that the input id mappings are correct.

Please let me know how I could solve this issue. Thanks in advance.

**Comment by SWivid** [comment]
the smallest corpus we have tried is 24 hours LJSpeech, and with batch_size_per_gpu 38400 (8gpus)
100K updates to get pretty good result

so not sure if 2 hours set will work to train from scratch, maybe merge some open-source 44khz corpus to increase the data amount?

**Comment by yuseungwoo** [comment]
Thanks for reply @SWivid. I'am the modeler using your model and I also have same question like @w11wo. 
Is the training command @w11wo is use is alright? README.md say the training using train.py.. 

@SWivid  Could you share the training command from the scratch showing the above "the smallest corpus we have tried is 24 hours LJSpeech, and with batch_size_per_gpu 38400 (8gpus) 100K updates to get pretty good result"  

Best regards

**Comment by yuseungwoo** [comment]
def clear_text(text):
    """Clean and prepare text by lowering the case and stripping whitespace."""
    return  [" "] + [syllabel for syllabel in text.lower().strip()]
    
  this is tokenizer I used

**Comment by yuseungwoo** [comment]
You mentioned that you achieved good results at 100k updates with a setting of batch_size_per_gpu 38400 (8 GPUs). At that time, what was the value of max_samples?

Based on the default settings in train.py, LJSpeech consists of 30 batches per epoch. Did you achieve good results after training up to 100k under these conditions?

In my case, if I set max_samples to 8 and grad_accumulation_steps to 8, the number of batches increases to 200, which is about 7 times more. Does that mean I need to train up to approximately 700k updates to achieve good sound quality?

**Comment by SWivid** [comment]
Hi @yuseungwoo , 

> max_samples?, mixed_precision:?

64, fp16

> if I set max_samples to 8 and grad_accumulation_steps to 8, the number of batches increases to 200, which is about 7 times more. Does that mean I need to train up to approximately 700k updates to achieve good sound quality?

Maybe not max_samples to 8, but only change the batchsize. 
If use 1/8 batchsize, grad_accumulation_steps to 8, and train to 100k*8=800k updates

**Comment by Mattias421** [comment]
Hi all, I am also interested in training on the LJSpeech dataset and have not managed to train a competitive model.

My general settings are:
model: F5TTS_Small
batch_size_per_gpu: 19200 
grad_accumulation_steps: 4
epochs: 4000

I'm training on 4 80GB A100s at fp16 precision. Specifically I'm training on the train split defined by NVIDIAs tacotron implementation.

After 4000 epochs / 400,000k steps, the model still fails to pronounce many sentences e.g. "The Warren Commission report by the president's commission on the assassination of President Kennedy"

It sounds like the alignment gets mixed up e.g. saying words like "assassikennedy"

Are there any tips on how to successfully train a model on LJSpeech?

**Comment by clumsyroot** [comment]
> @SWivid I fixed my parameters according to your comment.
> 
> exp_name = "F5TTS_Base" # F5TTS_Base | E2TTS_Base learning_rate = 7.5e-5
> 
> batch_size_per_gpu = 4800 # 8 GPUs, 8 * 38400 = 307200 batch_size_type = "frame" # "frame" or "sample" max_samples = 64 # max sequences per batch if use frame-wise batch_size. we set 32 for small models, 64 for base models grad_accumulation_steps = 8 # note: updates = steps / grad_accumulation_steps max_grad_norm = 1.0
> 
> epochs = 10000 # use linear decay, thus epochs control the slope num_warmup_updates = 20000 # warmup steps save_per_updates = 5000 # save checkpoint per steps last_per_steps = 5000 # save last checkpoint per steps

any updates ? I'm also working on training a small model with a limited dataset of around 100 hours. Curious to hear your progress. 😃

**Comment by ILG2021** [comment]
Hello, everyone, I am trying to train a 44k model from scratch too, I have about 10 hours data, 2000 batch size on a single 4080 GPU. after 250k updates, the voice become clear, but content is a mess. When go to 300k, voice become some noisy. The learning rate is 1e-13, gradient is nearly not update. I think the model can not learn the alignment.
When I use pretrain 22k model, 14k udpate will get a clear voice and accurate content at the same parameters.

**Comment by ILG2021** [comment]
> Hi all, I am also interested in training on the LJSpeech dataset and have not managed to train a competitive model.
> 
> My general settings are: model: F5TTS_Small batch_size_per_gpu: 19200 grad_accumulation_steps: 4 epochs: 4000
> 
> I'm training on 4 80GB A100s at fp16 precision. Specifically I'm training on the train split defined by NVIDIAs tacotron implementation.
> 
> After 4000 epochs / 400,000k steps, the model still fails to pronounce many sentences e.g. "The Warren Commission report by the president's commission on the assassination of President Kennedy"
> 
> It sounds like the alignment gets mixed up e.g. saying words like "assassikennedy"
> 
> Are there any tips on how to successfully train a model on LJSpeech?
> 
> (Alternatively I can train on 2 80GB H100s with grad_accumulation set to 4, I am currently testing which method is faster)

For LJSpeech, I don't think you need to train from scratch, it is nearly 24k, you can use the base pretrain to finetune, use a single GPU, may be 500k updates, you can get a nice model.

**Comment by Mattias421** [comment]
> > Hi all, I am also interested in training on the LJSpeech dataset and have not managed to train a competitive model.
> > My general settings are: model: F5TTS_Small batch_size_per_gpu: 19200 grad_accumulation_steps: 4 epochs: 4000
> > I'm training on 4 80GB A100s at fp16 precision. Specifically I'm training on the train split defined by NVIDIAs tacotron implementation.
> > After 4000 epochs / 400,000k steps, the model still fails to pronounce many sentences e.g. "The Warren Commission report by the president's commission on the assassination of President Kennedy"
> > It sounds like the alignment gets mixed up e.g. saying words like "assassikennedy"
> > Are there any tips on how to successfully train a model on LJSpeech?
> > (Alternatively I can train on 2 80GB H100s with grad_accumulation set to 4, I am currently testing which method is faster)
> 
> For LJSpeech, I don't think you need to train from scratch, it is nearly 24k, you can use the F5 TTS Base pretrain model to finetune, use a single GPU, may be 250k updates, you can get a nice model.
> 
> and matcha tts is trained on lj speech, you can use it directly, I think the sound is also very good. https://shivammehta25.github.io/Matcha-TTS/

Hi @ILG2021 , thanks for the advice! I plan on comparing the performance of F5TTS with Matcha-TTS, but fine-tuning seems like a good method too.
